mat/matrix.go

```diff
@@ -846,6 +846,28 @@ func normLapack(norm float64, aTrans bool) lapack.MatrixNorm {
 	}
 }
 
+// Rank returns rank of input matrix using singular value decomposition.
+// Give non-negative epsilon to determine singular values as zero.
+// If input epsilon is exactly 0, it is replaced with 1e-10, the default value.
+func Rank(a Matrix, epsilon float64) int {
+	if epsilon < 0 {
+		panic(ErrNegativeEpsilon)
+	}
+	if epsilon == 0 {
```

Comment:
I would think dlamchE or 2*dlamchE would be appropriate if this approach is being used (this I'm not convinced of though).

Reply:
Is that related to SVD implementation or something? (I'm not quite into it yet...)

Reply:
https://github.com/gonum/gonum/blob/a5f8f3a4840b76492571a7beed7958b420800458/lapack/gonum/lapack.go#L38-L39

However, I'm not still not convinced that doing this magically is the correct thing to do.

Reply:
I put the suggestion of a default value for the zero epsilon as a question, indicating that I also wasn't sure whether it's a good idea. Seeing it now in the code I'm not convinced either.

Reply:
I personally tested a little more: https://play.golang.org/p/V7HW_RAZxUr
It generates random float value matrix with given rank and apply this Rank() function.
Seeing returned singular value vectors, `1.0 / (1 << 53)` is too small and `1e-10` is enough for most cases. (you can change const values to test various cases)

Reply:
@vladimir-ch where can I see your suggestion?

Reply:
Please remove this.

Reply:
I should add, in [stat/mds](https://godoc.org/gonum.org/v1/gonum/stat/mds#TorgersonScaling) we have a similar problem [from a similar cause](https://github.com/gonum/gonum/issues/768). The approach there is to return the slice of Eigenvalues so the user can make an informed choice.

We already have the sigma vector, so perhaps we should just return that as well. @vladimir-ch @btracey?